packages/adapter-flagsmith/src/index.ts

```diff
@@ -0,0 +1,84 @@
+import type { Adapter } from 'flags';
+import flagsmith, {
+  IFlagsmithFeature,
+  IInitConfig,
+  IIdentity,
+  createFlagsmithInstance,
+} from 'flagsmith';
+
+type AdapterParams = {
+  key?: string;
+  projectID?: string;
+  dashboardURL?: string;
+} & IInitConfig;
+
+export function createFlagsmithAdapter<
+  ValueType extends IFlagsmithFeature,
+  EntitiesType extends IIdentity,
+>({
+  key: customKey,
+  environmentID,
+  projectID,
+  dashboardURL,
+  ...configParams
+}: AdapterParams): Adapter<ValueType, EntitiesType> {
+  async function initialize() {
+    const { identity, ...restConfigParams } = configParams;
+    const flagsmithInstance = createFlagsmithInstance();
+    await flagsmithInstance.init({ fetch: global.fetch, ...restConfigParams });
+    return flagsmithInstance;
+  }
+
+  return {
+    origin: (key: string) => {
+      const state = flagsmith.getState();
+      const flag = state?.flags?.[key];
+      return `${dashboardURL}/project/${projectID}/environment/${environmentID}/features/?feature=${flag?.id}`;
+    },
```

Comment:
It looks to me like this would be better provided through `getProviderData` which can list out all flags for a given host/project/environment,

see examples of provider/index.ts in other adapters, and move variables that are exclusively for enumerating origin there